### PR TITLE
Fix autotools scripts

### PR DESCRIPTION
**Pull request by PureTryOut**

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -1,15 +1,7 @@
...
-	`pkg-config --cflags glib-2.0` \
-	`pkg-config --cflags gconf-2.0` \
-	`pkg-config --cflags dbus-1` \
-	`pkg-config --cflags dbus-glib-1` \
-	-DLOCALEDIR=\"`pkg-config --variable=localedir osso-af-settings`\"
-
-hulda_LDFLAGS = \
-	`pkg-config --libs gconf-2.0 dbus-glib-1`
+hulda_CFLAGS= $(GLIB_CFLAGS) $(GCONF_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_CFLAGS)
+hulda_LDADD = $(GLIB_LIBS) $(GCONF_LIBS) $(DBUS_GLIB_LIBS) $(DBUS_LIBS)
```
Comment (freemangordon) [inline src/Makefile.am:7]: You're missing OSSO_CFLAGS/OSSO_LIBS. Also, I don;t see -DLOCALEDIR=\"`pkg-config --variable=localedir osso-af-settings`\" replacement

Comment (PureTryOut) [reply src/Makefile.am:7]: > `pkg-config --variable=localedir osso-af-settings`

Ah, that explains the last build error I had. I re-added that line, as I'm not sure how to put it in `configure.ac` instead. It builds successfully for me now.

**Comment by PureTryOut** [comment]
Everything builds correctly for me now.

**Comment by MerlijnWajer** [comment]
@freemangordon - looks ok to me. What do you think?

File: configure.ac [hunk 2]
```diff
@@ -3,4 +3,27 @@ AM_INIT_AUTOMAKE(ke-recv-extra, 0.2)
...
+AC_SUBST(GCONF_FLAGS)
+AC_SUBST(GCONF_LIBS)
+
+PKG_CHECK_MODULES(DBUS, dbus-1)
+AC_SUBST(DBUS_CFLAGS)
+AC_SUBST(DBUS_LIBS)
+
+PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1)
+AC_SUBST(DBUS_GLIB_CFLAGS)
+AC_SUBST(DBUS_GLIB_LIBS)
```
Comment (spinal84) [inline configure.ac:24]: All these lines can be replaced with a single:
PKG_CHECK_MODULES([DEPS], [libosso glib-2.0 gconf-2.0 dbus-1 dbus-glib-1])

File: configure.ac [hunk 3]
```diff
@@ -3,4 +3,27 @@ AM_INIT_AUTOMAKE(ke-recv-extra, 0.2)
...
+
+LOCALEDIR=`pkg-config --variable=localedir osso-af-settings`
+AC_SUBST(LOCALEDIR)
+
```
Comment (spinal84) [inline configure.ac:28]: This can be done in Makefile.am

File: src/Makefile.am [hunk 4]
```diff
@@ -1,15 +1,9 @@
 sbin_PROGRAMS = hulda
 
-INCLUDES = \
-	`pkg-config --cflags libosso` \
-	`pkg-config --cflags glib-2.0` \
-	`pkg-config --cflags gconf-2.0` \
-	`pkg-config --cflags dbus-1` \
-	`pkg-config --cflags dbus-glib-1` \
-	-DLOCALEDIR=\"`pkg-config --variable=localedir osso-af-settings`\"
+INCLUDES = -DLOCALEDIR=\"$(LOCALEDIR)\"
```
Comment (spinal84) [inline src/Makefile.am:3]: Correct value is:
AM_CPPFLAGS = -DLOCALEDIR=\"$(LOCALEDIR)\"

File: src/Makefile.am [hunk 5]
```diff
@@ -1,15 +1,9 @@
...
 
-hulda_LDFLAGS = \
-	`pkg-config --libs gconf-2.0 dbus-glib-1`
+hulda_CFLAGS= $(GLIB_CFLAGS) $(GCONF_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_CFLAGS) $(OSSO_CFLAGS)
+hulda_LDADD = $(GLIB_LIBS) $(GCONF_LIBS) $(DBUS_GLIB_LIBS) $(DBUS_LIBS) $(OSSO_LIBS)
```
Comment (spinal84) [inline src/Makefile.am:6]: Can be replaced with simple:
AM_CFLAGS = $(DEPS_CFLAGS)
LDADD = $(DEPS_LIBS)

File: configure.ac (hunk 2 above)
Comment (MerlijnWajer) [reply configure.ac:24]: I wouldn't necessarily lump them together. I think it's not a bad practice to separate them.

File: configure.ac (hunk 3 above)
Comment (MerlijnWajer) [reply configure.ac:28]: I think we do this on configure.ac in most of our repos, don't we?

Comment (spinal84) [reply configure.ac:28]: https://github.com/maemo-leste/ke-recv/blob/30f2486b1aa66c5c0b02d0884e3a2754ff39ed73/src/Makefile.am#L14

File: configure.ac (hunk 2 above)
Comment (spinal84) [reply configure.ac:24]: There is no need to inflate this file.

Comment (MerlijnWajer) [reply configure.ac:24]: We have this everywhere, and I think this is the usual way to do this?

https://github.com/maemo-leste/libhildon/blob/master/configure.ac#L42
https://github.com/maemo-leste/libhildondesktop/blob/master/configure.ac#L99

etc.

Comment (spinal84) [reply configure.ac:24]: Trying to make copy-paste doesn't seem a good approach to me. We maintain pretty old code base. Much has changed in the last 10-15 years.
And we should avoid using overcomplicated things when they can be simplified.
You already could see such examples:
Old approach: https://github.com/community-ssu/ke-recv/blob/master/debian/rules
New approach: https://github.com/spinal84/ke-recv/blob/kbd-slide/debian/rules
If we want to move forward in Maemo Leste development, we should simplify things, not overcomplicate them without a need.

File: configure.ac (hunk 3 above)
Comment (freemangordon) [reply configure.ac:28]: configure.ac is the right place for it, not Makefile.am

File: configure.ac (hunk 2 above)
Comment (freemangordon) [reply configure.ac:24]: I don't think there is a 'standard' way of doing this, both ways are fine IMO. Not to say that when more targets are build in a single project, using separate defines for pgkconfig flags is a must as usually includes/libs for those targets do not match. So I don't see any problem here.

File: src/Makefile.am (hunk 5 above)
Comment (freemangordon) [reply src/Makefile.am:6]: Yes, it can be replaced, however it is fine as it is as well.

**Comment by spinal84** [comment]
You guys should take a look at https://en.wikipedia.org/wiki/KISS_principle

**Comment by spinal84** [comment]
@MerlijnWajer, @freemangordon, please let me know if KISS is something to be ignored in Maemo Leste development.

File: configure.ac (hunk 3 above)
Comment (spinal84) [reply configure.ac:28]: Agree. Will do the same in ke-recv.

Comment (spinal84) [reply configure.ac:28]: Please check commit  https://github.com/maemo-leste/ke-recv/commit/9c73ad50a8e37d87bfdd20be1664e8a4a3cb4041
Using AC_DEFINE_UNQUOTED macro seems to be more appropriate here.

**Comment by MerlijnWajer** [comment]
I'm quite familiar with KISS principles, and I think they generally make a lot of sense (I apply them at my job a lot). But it's not something you can throw around to have things your way when you want it. Does KISS mean that we don't do translations for Maemo? Does it mean we don't want DBUS? There are many ways to interpret it. I personally don't care that much about how you do it, and I believe there are simplicity arguments to be made either way.

The more lengthy and explicit way is very simple, and clear to read. The DEPS one has everything stacked in one call, and might give a weird(er) error in `./configure` if it doesn't exist (didn't check this), on the other hand, I've made typos in `Makefile.am` when writing out all the `_CFLAGS` and `_LIBS`.

Honestly, I think that we should accept patches from people in either style, and realise that: "this really doesn't matter that much", and pick whichever we think works for us (so I mean: allow both, and apply standard when you work on it). If you want it written one way, just change it after the fact and I won't change/revert the change. There's probably a reason that for some build systems, there are many ways to things. Let's just for the fact remember that we mostly deal with Makefiles and autotools, and not all the other stuff, like Cmake, Scons, etc, etc... ;)

**Comment by parazyd** [comment]
Merged what I could in https://github.com/maemo-leste/ke-recv-extra/commit/d5f6eae97e1ecf2ac98c4a941d80e617639d9d79

**Pull request closed without merging**